src/main/java/christmas/controller/ChristmasController.java

```diff
@@ -0,0 +1,216 @@
+package christmas.controller;
+
+import static christmas.static_class.CommonStatic.NONE_NAME;
+
+import christmas.domain.DiscountDetailDto;
+import christmas.domain.GiftMenuDto;
+import christmas.domain.MoneyManagement;
+import christmas.domain.date.VisitDate;
+import christmas.domain.discount.EventBadge;
+import christmas.domain.menu.component.Name;
+import christmas.domain.menu.component.Price;
+import christmas.domain.order.Order;
+import christmas.domain.order.Orders;
+import christmas.exception.outside.DateUnitException;
+import christmas.exception.outside.order.OrderParentsException;
+import christmas.view.InputView;
+import christmas.view.OutputView;
+import java.util.List;
+
+public class ChristmasController {
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final VisitDate visitDate;
+    private final Orders orders;
+    private final MoneyManagement moneyManagement;
+
+    public ChristmasController(final InputView inputView, final OutputView outputView) {
+        outputView.introduce();
+        this.inputView = inputView;
+        this.outputView = outputView;
+        visitDate = initVisitDate();
+        orders = initOrders();
+        previewDate(outputView);
+        moneyManagement = initMoneyManagement();
+    }
+
+    private void previewDate(final OutputView outputView) {
+        int date = visitDate.getDate();
+        outputView.preview(date);
+    }
+
+    private VisitDate initVisitDate() {
+        try {
+            String userDate = inputView.readDate();
+            return new VisitDate(userDate);
+        } catch (DateUnitException de) {
+            outputView.printMessage(de.getMessage());
+            return initVisitDate();
+        }
+    }
+
+    private Orders initOrders() {
+        try {
+            String userMenu = inputView.readMenu();
+            return new Orders(userMenu);
+        } catch (OrderParentsException oe) {
+            outputView.printMessage(oe.getMessage());
+            return initOrders();
+        }
+    }
```

Comment:
컨트롤러에 존재하는 중복되는 `try-catch(재귀호출)`의 흐름은 함수형 인터페이스를 사용해 추출해 낼 수 있습니다!
아래는 임의로 만들어본 코드입니다!

(저의 경우 커스텀 예외를 사용하지는 않았었고, 아래 코드는준형님의 코드에 최대한 적용하기 위해 구글링을 하며 만든 코드입니다. 안정적인 코드는 아닐 수 있기에 이런 방법도 있구나, 라는 정도로만 참고 해 주시길 바랍니다!)
```suggestion
    private VisitDate initVisitDate() {
        return getOrRetry(() -> {
            String userDate = inputView.readDate();
            return new VisitDate(userDate);
        }, DateUnitException.class);
    }

    private Orders initOrders() {
        return getOrRetry(() -> {
            String userMenu = inputView.readMenu();
            return new Orders(userMenu);
        }, OrderParentsException.class);
    }

    private <T, X extends IllegalArgumentException> T getOrRetry(
            Supplier<T> supplier,
            Class<X> expectedException){
        try{
            return supplier.get();
        } catch (IllegalArgumentException e){
            if(e.getClass().equals(expectedException)){
                outputView.printMessage(e.getMessage());
                return supplier.get();
            }
            throw e;
        }
    }
```

Reply:
다시 한 번 느끼지만 장수님은 함수형 인터페이스, Stream 등 Java8 에 도입된 것들의 효율을 극대화해서 사용하시는 것 같아요. 정말 감사합니다!

Reply:
모던 자바 인 액션 책을 너무 재미있게 읽은 탓인 것 같습니다...ㅎㅎ
혹시 아직 읽어보시지 않으셨다면 꼭 한번 읽어보시길 추천드립니다!